Core/Src/main.c

```diff
@@ -34,6 +36,17 @@
 
 /* Private define ------------------------------------------------------------*/
 /* USER CODE BEGIN PD */
+
+const int ADC_TIMEOUT = 1000;   // 48 clock edges at 50 Hz is 960ms, set 1000ms to be safe
```

Comment:
why not uint16_t? In the embedded world, knowing the amount of space you are taking is very important. also, it adds readability since you would know the range of the uint16_t. for int, the size of it depends on the platform.